x/wasm/keeper/msg_dispatcher_test.go

```diff
@@ -184,6 +184,24 @@ func TestDispatchSubmessages(t *testing.T) {
 			expData:    []byte("myReplyData:1"),
 			expCommits: []bool{false, false},
 		},
+		"multiple msg - last reply can be empty to overwrite": {
+			msgs: []wasmvmtypes.SubMsg{{ID: 1, ReplyOn: wasmvmtypes.ReplyError}, {ID: 2, ReplyOn: wasmvmtypes.ReplyError}},
+			replyer: &mockReplyer{
+				replyFn: func(ctx sdk.Context, contractAddress sdk.AccAddress, reply wasmvmtypes.Reply) ([]byte, error) {
+					if reply.ID == 2 {
+						return []byte{}, nil
+					}
+					return []byte("myReplyData:1"), nil
+				},
+			},
+			msgHandler: &wasmtesting.MockMessageHandler{
+				DispatchMsgFn: func(ctx sdk.Context, contractAddr sdk.AccAddress, contractIBCPortID string, msg wasmvmtypes.CosmosMsg) (events []sdk.Event, data [][]byte, err error) {
+					return nil, nil, errors.New("my error")
+				},
+			},
+			expData:    []byte{},
```

Comment:
I am not sure this is desired.
I see the power of explicitly clearing any data.
But when do we want that?

I have seen way too many mix-ups between `[]byte{}` and `nil` in Go, especially when it gets serialized and later deserialized (they both go to `nil` I believe). I would rather just say "no bytes means don't change".

Is there an issue to change this functionality? I would rather handle the two the same in this PR and we can open up a new Issue to discuss this one case (and not block this PR on it).

Reply:
I was puzzled by nil/ empty as well and then re-read the https://github.com/CosmWasm/cosmwasm/blob/main/SEMANTICS.md#handling-the-reply doc. This point is not mentioned explicit but I was interpreting Some/None as the differentiator. `Some(EmptyString)` would overwrite the result.
Current implementation is more complete but I don't have a strong opinion. I will address your concern

Reply:
I am not so confident that `None` / `Some(vec![])` / `Some(vec![1, 2, 3])` will be converted to
`nil` / `[]byte{}` / `[]byte{1, 2, 3}`. 

This may be the case, but having a JSON encoder/decoder between languages makes me skeptical.
Maybe @webmaster128 could weigh in if this is checked full stack?
Otherwise, I would go for the safe route and add a FIXME to verify if this gets handles properly between wasm and go

Reply:
👌  safe route is preferred